src/FormBuilder/resources/js/form-builder/src/stores/editor-state/index.tsx

```diff
@@ -0,0 +1,28 @@
+import {createContext, ReactNode, useContext, useReducer} from 'react';
+import editorStateReducer, {setEditorMode} from './reducer';
+
+export interface EditorState {
+    mode: string;
+}
+
+const StoreContext = createContext(null);
+StoreContext.displayName = 'EditorState';
+
+const StoreContextDispatch = createContext(null);
+StoreContextDispatch.displayName = 'EditorStateDispatch';
+
+const EditorStateProvider = ({initialState, children}: { initialState: EditorState; children: ReactNode }) => {
+    const [state, dispatch] = useReducer(editorStateReducer, initialState);
+
+    return (
+        <StoreContext.Provider value={state}>
+            <StoreContextDispatch.Provider value={dispatch}>{children}</StoreContextDispatch.Provider>
+        </StoreContext.Provider>
+    );
+};
+
+const useEditorState = () => useContext<EditorState>(StoreContext);
+
+const useEditorStateDispatch = () => useContext(StoreContextDispatch);
+
+export {EditorStateProvider, useEditorState, useEditorStateDispatch, setEditorMode};
```

Comment:
Will more data be added to this context in the future? If not, we can replace the complexity of a reducer with a simpler ﻿`useState`.

Just to fulfill my curiosity, why do we create two contexts here instead of passing both values as `﻿value={{state, dispatch}}`?

Reply:
Using the store allows us to use hooks in deeply nested components without prop drilling the state.

As far as the specifics of how this store is laid out... I copy/pasted @jonwaldstein's form state store. 😎